core/modules/replica/WorkerServer.h

```diff
@@ -0,0 +1,152 @@
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+// Forward declarations
+class WorkerInfo;
+class WorkerRequestFactory;
+
+/**
+  * Class WorkerServer is used for handling incomming connections to
+  * the worker replication service. Only one instance of this class is
+  * allowed per a thread.
+  */
+class WorkerServer
+    : public std::enable_shared_from_this<WorkerServer>  {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<WorkerServer> Ptr;
+
+    /**
+     * Static factory method is needed to prevent issue with the lifespan
+     * and memory management of instances created otherwise (as values or via
+     * low-level pointers).
+     *
+     * @param serviceProvider - for configuration, etc. services
+     * @param requestFactory  - the factory of requests
+     * @workerName            - the name of a worker this instance represents
+     *
+     * @return pointer to the new object created by the factory
+     */
+    static Ptr create(ServiceProvider::Ptr const& serviceProvider,
+                      WorkerRequestFactory& requestFactory,
+                      std::string const& workerName);
+
+    // Default construction and copy semantics are prohibited
+
+    WorkerServer() = delete;
+    WorkerServer(WorkerServer const&) = delete;
+    WorkerServer& operator=(WorkerServer const&) = delete;
+
+    ~WorkerServer() = default;
+
+    /// @return the name of a worker this server runs for
+    std::string const& worker() const { return _workerName; }
+
+    /**
+     * The processor object can be used for detailed monitoring of
+     * the on-going activities and statistics collection if needed.
+     *
+     * @return reference to the processor
+     */
+    WorkerProcessor const& processor() const { return _processor; }
+
+    /**
+     * Begin listening for and processing incoming connections
+     */
+    void run();
+
+private:
+
+    /**
+     * Construct the server with the specified configuration.
+     *
+     * @param serviceProvider - for configuration, etc. services
+     * @param requestFactory  - the factory of requests
+     * @workerName            - the name of a worker this instance represents
+     */
+    WorkerServer(ServiceProvider::Ptr const& serviceProvider,
+                 WorkerRequestFactory& requestFactory,
+                 std::string const& workerName);
+
+    /**
+     * Begin (asynchrnonously) accepting connection requests.
+     */
+    void beginAccept();
+    
+    /**
+     * Handle a connection request once it's detected. The rest of
+     * the comunication will be forewarded to the connection object
+     * specified as a parameter of the method.
+     *
+     * @param connection - object responsible for communications with a client
+     * @param ec         - error condition to be checked for
+     */
+    void handleAccept(WorkerServerConnection::Ptr const& connection,
+                      boost::system::error_code const& ec);
```

Comment:
passing the enum by const& is unnecessary, but AFAIK neither does it hurt. But you may consider passing by value.

Reply:
error_code is not a enum. It's a class.